[soci] Build failure: vcpkg install soci[mysql]. However, it has manual workaround and can be manually built

**Host Environment**
- OS: Linux (Ubuntu 20.04) and macOS (both Mojave and Catalina) 
- Compiler: GNU 9.3.0


**********************************

**To Reproduce**
Steps to reproduce the behavior:
./vcpkg install soci[mysql]

********************************

**Failure logs** 
```
root@ubuntu:/opt/vcpkg# ./vcpkg install soci[mysql]
Computing installation plan...
The following packages will be built and installed:
    soci[core,mysql]:x64-linux -> 4.0.1#1
Detecting compiler hash for triplet x64-linux...
Could not locate cached archive: /root/.cache/vcpkg/archives/95/950c8efbbb03febbb50a806e1f6dc5f35f09ee37.zip
Starting package 1/1: soci:x64-linux
Building package soci[core,mysql]:x64-linux...
-- Using cached /opt/vcpkg/downloads/SOCI-soci-334cc55d9fa7b42d7214a8533a246d637bc92899.tar.gz
-- Cleaning sources at /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /opt/vcpkg/downloads/SOCI-soci-334cc55d9fa7b42d7214a8533a246d637bc92899.tar.gz
-- Using source at /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean
-- Configuring x64-linux-rel
-- Building x64-linux-rel
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:144 (message):
    Command failed: /opt/vcpkg/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/bin/cmake --build . --config Release --target install -- -v -j5
    Working Directory: /opt/vcpkg/buildtrees/soci/x64-linux-rel
    See logs for more information:
      /opt/vcpkg/buildtrees/soci/install-x64-linux-rel-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:96 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:27 (vcpkg_build_cmake)
  ports/soci/portfile.cmake:50 (vcpkg_install_cmake)
  scripts/ports.cmake:131 (include)


Error: Building package soci:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: soci:x64-linux
  Vcpkg version: 2021-01-13-unknownhash

**********************************************************************
Additionally, attach any relevant sections from the log files above.
```
From log file:
/opt/vcpkg/buildtrees/soci/install-x64-linux-rel-out.log
```
FAILED: src/backends/mysql/CMakeFiles/soci_mysql_static.dir/blob.cpp.o
/usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I\"/opt/vcpkg/installed/x64-linux/include/mysql\" -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/blob.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/blob.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/blob.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/blob.cpp
In file included from /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/blob.cpp:10:
/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/soci/mysql/soci-mysql.h:31:10: fatal error: mysql.h: No such file or directory
   31 | #include <mysql.h> // MySQL Client
      |          ^~~~~~~~~
compilation terminated.
```
**************************************************************************
**Additional context**Workaround**
If I modify the CMakeCache.txt and re-run manually the command:
```
cd /opt/vcpkg/buildtrees/soci/x64-linux-rel
vim CMakeCache.txt
```
replace this line:
`MYSQL_INCLUDE_DIR:PATH=/opt/vcpkg/buildtrees/soci/x64-linux-rel/"/opt/vcpkg/installed/x64-linux/include/mysql"`

with this line:
`MYSQL_INCLUDE_DIR:PATH=/opt/vcpkg/installed/x64-linux/include/mysql`

Then run the command manually, then it works:

`/opt/vcpkg/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/bin/cmake --build . --config Release --target install -- -v -j5`

Result:
```
root@ubuntu:/opt/vcpkg/buildtrees/soci/x64-linux-rel# /opt/vcpkg/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/bin/cmake --build . --config Release --target install -- -v -j5
[0/1] /opt/vcpkg/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/bin/cmake --regenerate-during-build -S/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean -B/opt/vcpkg/buildtrees/soci/x64-linux-rel
CMake Deprecation Warning at CMakeLists.txt:13 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

  - Configuring SOCI: 
-- SOCI_VERSION                             = 4.0.1 
-- SOCI_ABI_VERSION                         = 4.0 
-- SOCI_SHARED                              = 0 
-- SOCI_STATIC                              = 1 
-- SOCI_TESTS                               = OFF 
-- SOCI_ASAN                                = OFF 
-- SOCI_CXX11                               = ON 
-- LIB_SUFFIX                               = 64 
-- Looking for SOCI dependencies: 
-- Threads: 
-- CMAKE_THREAD_LIBS_INIT                   = -lpthread 
-- Boost: disabled, since WITH_BOOST=OFF 
-- MySQL: 
-- Found MySQL: /opt/vcpkg/installed/x64-linux/include/mysql, /opt/vcpkg/installed/x64-linux/lib/libmysqlclient.a
-- MySQL Embedded not found.
-- MYSQL_INCLUDE_DIR                        = /opt/vcpkg/installed/x64-linux/include/mysql 
-- MYSQL_LIBRARIES                          = /opt/vcpkg/installed/x64-linux/lib/libmysqlclient.a 
-- ODBC: disabled, since WITH_ODBC=OFF 
-- Oracle: disabled, since WITH_ORACLE=OFF 
-- PostgreSQL: disabled, since WITH_POSTGRESQL=OFF 
-- SQLite3: disabled, since WITH_SQLITE3=OFF 
-- Firebird: disabled, since WITH_FIREBIRD=OFF 
-- DB2: disabled, since WITH_DB2=OFF 
-- Configuring SOCI core library: 
-- SOCI_CORE_TARGET                         = soci_core 
-- SOCI_CORE_TARGET_OUTPUT_NAME             = soci_core 
-- SOCI_CORE_DEPS_LIBS                      = -lpthread /usr/lib/x86_64-linux-gnu/libdl.so 
-- SOCI_CORE_INCLUDE_DIRS                   = /opt/vcpkg/buildtrees/soci/x64-linux-rel /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include /opt/vcpkg/buildtrees/soci/x64-linux-rel/include /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private /opt/vcpkg/buildtrees/soci/x64-linux-rel/src/core 
-- WITH_BOOST                               = OFF 
-- COMPILE_DEFINITIONS                      = SOCI_ABI_VERSION="4.0" HAVE_DL=1 SOCI_LIB_PREFIX="libsoci_" SOCI_LIB_SUFFIX=".so" SOCI_DEBUG_POSTFIX="" 
-- 
-- Configuring SOCI backend libraries: 
-- Found MySQL: /opt/vcpkg/installed/x64-linux/include/mysql, /opt/vcpkg/installed/x64-linux/lib/libmysqlclient.a
-- MySQL Embedded not found.
-- Empty - SOCI backend skeleton for new backends development 
-- SOCI_EMPTY                               = ON 
-- SOCI_EMPTY_TARGET                        = soci_empty 
-- SOCI_EMPTY_OUTPUT_NAME                   = soci_empty 
-- SOCI_EMPTY_COMPILE_DEFINITIONS           = SOCI_ABI_VERSION="4.0" HAVE_DL=1 
-- SOCI_EMPTY_INCLUDE_DIRECTORIES           = /opt/vcpkg/buildtrees/soci/x64-linux-rel /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include /opt/vcpkg/buildtrees/soci/x64-linux-rel/include /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private 
-- MySQL - SOCI backend for MySQL 
-- SOCI_MYSQL                               = ON 
-- SOCI_MYSQL_TARGET                        = soci_mysql 
-- SOCI_MYSQL_OUTPUT_NAME                   = soci_mysql 
-- SOCI_MYSQL_COMPILE_DEFINITIONS           = SOCI_ABI_VERSION="4.0" HAVE_DL=1 
-- SOCI_MYSQL_INCLUDE_DIRECTORIES           = /opt/vcpkg/buildtrees/soci/x64-linux-rel /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include /opt/vcpkg/buildtrees/soci/x64-linux-rel/include /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private /opt/vcpkg/installed/x64-linux/include/mysql 
-- 
-- 
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/vcpkg/buildtrees/soci/x64-linux-rel
[1/14] : && /opt/vcpkg/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/bin/cmake -E rm -f lib/libsoci_empty.a && /usr/bin/ar qc lib/libsoci_empty.a  src/backends/empty/CMakeFiles/soci_empty_static.dir/blob.cpp.o src/backends/empty/CMakeFiles/soci_empty_static.dir/factory.cpp.o src/backends/empty/CMakeFiles/soci_empty_static.dir/row-id.cpp.o src/backends/empty/CMakeFiles/soci_empty_static.dir/session.cpp.o src/backends/empty/CMakeFiles/soci_empty_static.dir/standard-into-type.cpp.o src/backends/empty/CMakeFiles/soci_empty_static.dir/standard-use-type.cpp.o src/backends/empty/CMakeFiles/soci_empty_static.dir/statement.cpp.o src/backends/empty/CMakeFiles/soci_empty_static.dir/vector-into-type.cpp.o src/backends/empty/CMakeFiles/soci_empty_static.dir/vector-use-type.cpp.o && /usr/bin/ranlib lib/libsoci_empty.a && :
[2/14] : && /opt/vcpkg/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/bin/cmake -E rm -f lib/libsoci_core.a && /usr/bin/ar qc lib/libsoci_core.a  src/core/CMakeFiles/soci_core_static.dir/backend-loader.cpp.o src/core/CMakeFiles/soci_core_static.dir/blob.cpp.o src/core/CMakeFiles/soci_core_static.dir/common.cpp.o src/core/CMakeFiles/soci_core_static.dir/connection-parameters.cpp.o src/core/CMakeFiles/soci_core_static.dir/connection-pool.cpp.o src/core/CMakeFiles/soci_core_static.dir/error.cpp.o src/core/CMakeFiles/soci_core_static.dir/into-type.cpp.o src/core/CMakeFiles/soci_core_static.dir/logger.cpp.o src/core/CMakeFiles/soci_core_static.dir/once-temp-type.cpp.o src/core/CMakeFiles/soci_core_static.dir/prepare-temp-type.cpp.o src/core/CMakeFiles/soci_core_static.dir/procedure.cpp.o src/core/CMakeFiles/soci_core_static.dir/ref-counted-prepare-info.cpp.o src/core/CMakeFiles/soci_core_static.dir/ref-counted-statement.cpp.o src/core/CMakeFiles/soci_core_static.dir/row.cpp.o src/core/CMakeFiles/soci_core_static.dir/rowid.cpp.o src/core/CMakeFiles/soci_core_static.dir/session.cpp.o src/core/CMakeFiles/soci_core_static.dir/soci-simple.cpp.o src/core/CMakeFiles/soci_core_static.dir/statement.cpp.o src/core/CMakeFiles/soci_core_static.dir/transaction.cpp.o src/core/CMakeFiles/soci_core_static.dir/use-type.cpp.o src/core/CMakeFiles/soci_core_static.dir/values.cpp.o && /usr/bin/ranlib lib/libsoci_core.a && :
[3/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/common.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/common.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/common.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/common.cpp
[4/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/factory.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/factory.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/factory.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/factory.cpp
[5/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/blob.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/blob.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/blob.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/blob.cpp
[6/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/row-id.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/row-id.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/row-id.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/row-id.cpp
[7/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/session.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/session.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/session.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/session.cpp
[8/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/standard-into-type.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/standard-into-type.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/standard-into-type.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/standard-into-type.cpp
[9/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/statement.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/statement.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/statement.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/statement.cpp
[10/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/standard-use-type.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/standard-use-type.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/standard-use-type.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/standard-use-type.cpp
[11/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/vector-into-type.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/vector-into-type.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/vector-into-type.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/vector-into-type.cpp
[12/14] /usr/bin/c++ -DHAVE_DL=1 -DSOCI_ABI_VERSION=\"4.0\" -I. -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include -Iinclude -I/opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/include/private -I/opt/vcpkg/installed/x64-linux/include/mysql -fPIC -pedantic  -Wno-error=parentheses -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Woverloaded-virtual -Wredundant-decls -Wno-long-long -std=c++11  -Wno-variadic-macros -O3 -DNDEBUG -std=gnu++11 -MD -MT src/backends/mysql/CMakeFiles/soci_mysql_static.dir/vector-use-type.cpp.o -MF src/backends/mysql/CMakeFiles/soci_mysql_static.dir/vector-use-type.cpp.o.d -o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/vector-use-type.cpp.o -c /opt/vcpkg/buildtrees/soci/src/637bc92899-f2610ffee1.clean/src/backends/mysql/vector-use-type.cpp
[13/14] : && /opt/vcpkg/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/bin/cmake -E rm -f lib/libsoci_mysql.a && /usr/bin/ar qc lib/libsoci_mysql.a  src/backends/mysql/CMakeFiles/soci_mysql_static.dir/blob.cpp.o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/common.cpp.o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/factory.cpp.o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/row-id.cpp.o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/session.cpp.o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/standard-into-type.cpp.o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/standard-use-type.cpp.o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/statement.cpp.o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/vector-into-type.cpp.o src/backends/mysql/CMakeFiles/soci_mysql_static.dir/vector-use-type.cpp.o && /usr/bin/ranlib lib/libsoci_mysql.a && :
[13/14] cd /opt/vcpkg/buildtrees/soci/x64-linux-rel && /opt/vcpkg/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/bin/cmake -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/soci-config.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/backend-loader.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/bind-values.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/blob-exchange.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/blob.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/boost-fusion.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/boost-gregorian-date.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/boost-optional.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/boost-tuple.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/callbacks.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/column-info.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/connection-parameters.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/connection-pool.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/error.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/exchange-traits.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/into-type.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/into.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/logger.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/noreturn.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/once-temp-type.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/prepare-temp-type.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/procedure.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/query_transformation.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/ref-counted-prepare-info.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/ref-counted-statement.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/row-exchange.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/row.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/rowid-exchange.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/rowid.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/rowset.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/session.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/soci-backend.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/soci-platform.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/soci-simple.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/soci.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/statement.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/transaction.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/type-conversion-traits.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/type-conversion.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/type-holder.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/type-ptr.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/type-wrappers.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/unsigned-types.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/use-type.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/use.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/values-exchange.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/values.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/version.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/lib/libsoci_core.a
-- Installing: /opt/vcpkg/packages/soci_x64-linux/cmake/SOCI.cmake
-- Installing: /opt/vcpkg/packages/soci_x64-linux/cmake/SOCI-release.cmake
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/empty/soci-empty.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/lib/libsoci_empty.a
-- Up-to-date: /opt/vcpkg/packages/soci_x64-linux/cmake/SOCI.cmake
-- Up-to-date: /opt/vcpkg/packages/soci_x64-linux/cmake/SOCI-release.cmake
-- Installing: /opt/vcpkg/packages/soci_x64-linux/include/soci/mysql/soci-mysql.h
-- Installing: /opt/vcpkg/packages/soci_x64-linux/lib/libsoci_mysql.a
-- Up-to-date: /opt/vcpkg/packages/soci_x64-linux/cmake/SOCI.cmake
-- Up-to-date: /opt/vcpkg/packages/soci_x64-linux/cmake/SOCI-release.cmake
```
Summary:

vcpkg install soci[mysql] 

fails because there seems that MYSQL_INCLUDE_DIR doesn't get filled properly (the dependency is also installed by vcpkg) (ie. MYSQL_INCLUDE_DIR=/opt/vcpkg/buildtrees/soci/x64-linux-rel/"/opt/vcpkg/installed/x64-linux/include/mysql" instead of MYSQL_INCLUDE_DIR=/opt/vcpkg/installed/x64-linux/include/mysql)

Thanks,
Gabriel.

--- a/ports/soci/export-include-dirs.patch
+++ b/ports/soci/export-include-dirs.patch
@@ -1,0 +1,53 @@
+diff --git a/cmake/SociBackend.cmake b/cmake/SociBackend.cmake
+index 442d81d..ef56422 100644
+--- a/cmake/SociBackend.cmake
++++ b/cmake/SociBackend.cmake
+@@ -159,8 +159,13 @@ macro(soci_backend NAME)
+           ${THIS_BACKEND_HEADERS})
+ 
+         target_link_libraries(${THIS_BACKEND_TARGET}
++          PUBLIC
+           ${SOCI_CORE_TARGET}
+           ${THIS_BACKEND_DEPENDS_LIBRARIES})
++          
++        target_include_directories(${THIS_BACKEND_TARGET}
++          PUBLIC
++          ${THIS_BACKEND_DEPENDS_INCLUDE_DIRS})
+ 
+         if(WIN32)
+           set_target_properties(${THIS_BACKEND_TARGET}
+@@ -197,8 +202,14 @@ macro(soci_backend NAME)
+ 
+         # Still need to link the libraries for tests to work
+         target_link_libraries (${THIS_BACKEND_TARGET_STATIC}
++          PUBLIC
+           ${THIS_BACKEND_DEPENDS_LIBRARIES}
+         )
++        
++        target_include_directories(${THIS_BACKEND_TARGET_STATIC}
++          PUBLIC
++          ${THIS_BACKEND_DEPENDS_INCLUDE_DIRS}
++        )
+ 
+         set_target_properties(${THIS_BACKEND_TARGET_STATIC}
+           PROPERTIES
+diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
+index 3916cce..d537613 100644
+--- a/src/core/CMakeLists.txt
++++ b/src/core/CMakeLists.txt
+@@ -64,6 +64,7 @@ if (SOCI_SHARED)
+   target_include_directories(${SOCI_CORE_TARGET}
+       PUBLIC
+       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../include>
++      $<INSTALL_INTERFACE:include>
+   )
+ 
+ endif()
+@@ -96,6 +97,7 @@ if (SOCI_STATIC)
+   target_include_directories(${SOCI_CORE_TARGET_STATIC}
+       PUBLIC
+       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../../include>
++      $<INSTALL_INTERFACE:include>
+   )
+ 
+ endif()

--- a/ports/soci/fix-dependency-libmysql.patch
+++ b/ports/soci/fix-dependency-libmysql.patch
@@ -1,12 +1,13 @@
 diff --git a/cmake/modules/FindMySQL.cmake b/cmake/modules/FindMySQL.cmake
-index 1233568..6110c36 100644
+index 1233568..9a7cb61 100644
 --- a/cmake/modules/FindMySQL.cmake
 +++ b/cmake/modules/FindMySQL.cmake
-@@ -13,6 +13,8 @@
+@@ -13,6 +13,9 @@
  #
  # Redistribution and use is allowed according to the terms of the BSD license.
  # For details see the accompanying COPYING-CMAKE-SCRIPTS file.
 +find_package(libmysql REQUIRED)
++set(MySQL_FOUND 1)
 +return()
  
  include(CheckCXXSourceCompiles)

--- a/ports/soci/portfile.cmake
+++ b/ports/soci/portfile.cmake
@@ -51,7 +51,14 @@ vcpkg_install_cmake()
 vcpkg_copy_pdbs()
 vcpkg_fixup_cmake_targets(CONFIG_PATH cmake)
 # Correct the config file name
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/${PORT}/SOCI.cmake ${CURRENT_PACKAGES_DIR}/share/${PORT}/SOCI-config.cmake)
+file(RENAME ${CURRENT_PACKAGES_DIR}/share/${PORT}/SOCI.cmake ${CURRENT_PACKAGES_DIR}/share/${PORT}/SOCIConfig.cmake)
+
+if ("mysql" IN_LIST FEATURES)
+    vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/share/${PORT}/SOCIConfig.cmake
+        "# Create imported target SOCI::soci_mysql"
+        "\ninclude(CMakeFindDependencyMacro)\nfind_dependency(libmysql)\n# Create imported target SOCI::soci_mysql"
+    )
+endif()
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include ${CURRENT_PACKAGES_DIR}/debug/share)
 

--- a/ports/soci/portfile.cmake
+++ b/ports/soci/portfile.cmake
@@ -4,7 +4,9 @@ vcpkg_from_github(
     REF 334cc55d9fa7b42d7214a8533a246d637bc92899 #version 4.0.1 commit on 2020.10.19
     SHA512 b300b13f68347d78252812e09efffb1735072cf5019940da53366a5cdee997f4b8b03a584a87a95ba764b0a78640ad6eb4966b53f9156280cb452465607afbc7
     HEAD_REF master
-    PATCHES fix-dependency-libmysql.patch
+    PATCHES
+        fix-dependency-libmysql.patch
+        export-include-dirs.patch
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" SOCI_DYNAMIC)

--- a/ports/soci/portfile.cmake
+++ b/ports/soci/portfile.cmake
@@ -26,7 +26,7 @@ foreach(_feature IN LISTS ALL_FEATURES)
     endif()
 
     if(_feature MATCHES "mysql")
-        set(MYSQL_OPT -DMYSQL_INCLUDE_DIR="${CURRENT_INSTALLED_DIR}/include/mysql")
+        set(MYSQL_OPT -DMYSQL_INCLUDE_DIR=${CURRENT_INSTALLED_DIR}/include/mysql)
     endif()
 endforeach()
 

--- a/ports/soci/vcpkg.json
+++ b/ports/soci/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "soci",
-  "version-string": "4.0.1",
-  "port-version": 2,
+  "version": "4.0.1",
+  "port-version": 3,
   "description": "SOCI database access library",
   "homepage": "https://github.com/SOCI/soci",
   "features": {

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5450,7 +5450,7 @@
     },
     "soci": {
       "baseline": "4.0.1",
-      "port-version": 2
+      "port-version": 3
     },
     "socket-io-client": {
       "baseline": "1.6.1-1",

--- a/versions/s-/soci.json
+++ b/versions/s-/soci.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "6ea85617d08f43b3a3921c020ce74df74c8906c8",
+      "git-tree": "c3d0fa171afa65a852f8555b66558a9a6183bcf4",
       "version": "4.0.1",
       "port-version": 3
     },

--- a/versions/s-/soci.json
+++ b/versions/s-/soci.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "6ea85617d08f43b3a3921c020ce74df74c8906c8",
+      "version": "4.0.1",
+      "port-version": 3
+    },
+    {
       "git-tree": "8127c6d88c6b673066ea41b061dd79d600c442d4",
       "version-string": "4.0.1",
       "port-version": 2

--- a/versions/s-/soci.json
+++ b/versions/s-/soci.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "c3d0fa171afa65a852f8555b66558a9a6183bcf4",
+      "git-tree": "05767aa636348d9fb5a2507a2c348091cc9d7782",
       "version": "4.0.1",
       "port-version": 3
     },

